pandas/core/indexes/multi.py

```diff
@@ -1111,8 +1112,8 @@ def _engine(self):
         # Check the total number of bits needed for our representation:
         if lev_bits[0] > 64:
             # The levels would overflow a 64 bit uint - use Python integers:
-            return MultiIndexPyIntEngine(self.levels, self.codes, offsets)
-        return MultiIndexUIntEngine(self.levels, self.codes, offsets)
+            return MultiIndexPyIntEngine(list(self.levels), self.codes, offsets)
```

Comment:
I don't think should need to change this. should probably be using sequence in BaseMultiIndexCodesEngine.\_\_init__

from https://github.com/python/typeshed/blob/master/CONTRIBUTING.md#conventions

> avoid invariant collection types (list, dict) in argument positions, in favor of covariant types like Mapping or Sequence;